shared/fs/banner/index.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {Box, Text, Icon, Button} from '../../common-adapters'
+import {fileUIName, isMobile} from '../../constants/platform'
+import {globalStyles, globalMargins, globalColors} from '../../styles'
+
+type Props = {
+  kbfsEnabled: boolean,
+  inProgress: boolean,
+  showBanner: boolean,
+  path?: Types.Path,
+  onDismiss?: () => void,
+  onInstall: () => void,
+  onUninstall: () => void,
+  openInFileUI?: () => void,
+}
+
+const Banner = isMobile
+  ? () => <Box />
+  : ({kbfsEnabled, showBanner, onInstall, onUninstall, onDismiss, openInFileUI}: Props) => {
+      if (kbfsEnabled && !showBanner) {
+        return <Box />
+      }
+      const iconType = kbfsEnabled ? 'icon-fancy-finder-enabled-132-96' : 'icon-fancy-finder-132-96'
+      const bannerStyle = {
+        ...globalStyles.flexBoxRow,
+        backgroundColor: kbfsEnabled ? globalColors.green : globalColors.blue,
+        height: 176,
+        alignItems: 'center',
+        position: 'relative',
+      }
+      let bannerContent
+      if (kbfsEnabled) {
+        bannerContent = (
+          <Box style={globalStyles.flexBoxColumn}>
+            <Text type="Header" style={textStyle}>
+              {onDismiss && 'Yay!'} Keybase is {onDismiss && 'now'} enabled in your {fileUIName}.
```

Comment:
you have extra weird spaces if you do it this way (even if some rendering stuff collapses it). Like you want either
```
{'Yay! '}Keybase is {'now '}enabled
```
etc or even better just make one string with templates
```
{`${onDismiss ? 'Yay! ' : ''}Keybase is${onDismiss ? ' now' : ' '}enabled in your ${fileUIName}`}
```